### PR TITLE
[docs] Exclude unrelated files

### PR DESCRIPTION
**Pull request by giordano**
`apps/` and `post-processing/` are symlinks to the directories in our repository, with the result the website includes *all* the files in them.

### CODE REVIEW

**Comment by giordano** [comment]
Our website currently bundles _all_ files in `apps/` and `post-processing`, with the result the [entire website has a size of whopping 39.2 MiB](https://github.com/ukri-excalibur/excalibur-tests/actions/runs/8377880301).  Instead with this PR the entire website, including the preview for the current PR, is [39.9 MiB](https://github.com/ukri-excalibur/excalibur-tests/actions/runs/8379106337), which means the website in the preview is only 0.7 MiB, instead of 39.2 MiB.

**Comment by giordano** [comment]
An alternative is to selectively symlink the `README.md` files for each individual benchmark, but that'd add an extra step to adding a new benchmark.

**Comment by tkoskela** [comment]
The `Details` link on the Documentation deployed action seems to have broken

**Comment by giordano** [comment]
Right, because that's a Python script, not a bash one 😅  Will fix it.

**Comment by giordano** [comment]
Fixed, now the link should be correct again.  For reference, it's https://ukri-excalibur.github.io/excalibur-tests/preview/PR293/

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -61,29 +61,46 @@ jobs:
...
+          # cleanly update the website.
           if [[ ${{ github.event_name }} == 'pull_request' ]]; then
-              COMMIT_MESSAGE="Update docs preview from ${{ github.sha }} for PR #${{ github.event.number }}"
+              COMMIT_MESSAGE="Update docs preview from ${{ github.sha }}"
+              # Only delete files in the preview subdir, if any
+              git rm -rf "${PREVIEW_SUBDIR}"
           else
               COMMIT_MESSAGE="Update docs from ${{ github.sha }}"
+              for file in $(git ls-files); do
+                  if [[ x"${file}" != xpreview/* ]]; then
```
Comment (tkoskela) [inline .github/workflows/docs.yml:96]: I don't understand this bit. You want to delete all files in the repository, except ones starting with `preview/`? Is the `x` just enforcing that `preview` is at the start of the path?

Comment (giordano) [reply .github/workflows/docs.yml:96]: Yes, and yes. The leading "x" is a common shell trick to ensure the comparison is done correctly, autotools scripts use it extensively for example.

**Review by tkoskela — APPROVED** [review]
Thanks for spotting this before we bloated the git history too badly!

**Comment by giordano** [comment]
We might also want to squash the history of the gh-pages branch at some point, so that we forget about the files that had been checked in.

**Comment by giordano** [comment]
New website is only [691 KiB](https://github.com/ukri-excalibur/excalibur-tests/actions/runs/8389577101), and [this diff](https://github.com/ukri-excalibur/excalibur-tests/commit/bd1f87764dfc8e34e8afdaccd25b53a071f8a618) shows all the deleted files 🙂

**Pull request merged**